### PR TITLE
Inbox Badges

### PR DESCRIPTION
**Pull request by zachatrocity**

### CODE REVIEW

File: lib/stores/accounts_store.dart [hunk 1]
```diff
@@ -163,6 +165,16 @@ class AccountsStore extends ChangeNotifier {
...
   }
 
+  Future<void> checkNotifications(UserData? userData) async {
+    notificationCount = await LemmyApiV3(userData!.instanceHost)
+        .run(GetUnreadCount(
+          auth: userData.jwt.raw,
+        ))
+        .then((e) => e.mentions + e.replies + e.privateMessages);
+
+    notifyListeners();
```
Comment (shocklateboy92) [inline lib/stores/accounts_store.dart:175]: All the others have a `return save();`? 
Do you need it here?

**Review by shocklateboy92 — CHANGES_REQUESTED** [review]

File: lib/pages/home_page.dart [hunk 2]
```diff
@@ -39,6 +41,12 @@ class HomePage extends HookWidget {
       return null;
     }, [theme.scaffoldBackgroundColor]);
 
+    useEffect(() {
+      accStore.checkNotifications(accStore.defaultUserData);
+
+      return null;
+    }, [accStore.defaultInstanceHost, currentTab]);
```
[the remark this answers is not in the document]
Comment (zachatrocity) [reply lib/pages/home_page.dart:48]: Well I want it to trigger when the tab changes. Does the key list have to be used within the use effect? I'm coming from a react world

File: lib/stores/accounts_store.dart (hunk 1 above)
Comment (zachatrocity) [reply lib/stores/accounts_store.dart:175]: No cause I didnt care about serializing notification count in prefs

File: lib/pages/home_tab.dart [hunk 3]
```diff
@@ -247,10 +247,22 @@ class HomeTab extends HookWidget {
...
-                ),
+                leading: accStore.notificationCount != null &&
+                        accStore.notificationCount! > 0
+                    ? Badge(
+                        offset: const Offset(-5, 5),
+                        label: Text(accStore.notificationCount!.toString()),
+                        child: IconButton(
+                          icon: const Icon(Icons.notifications),
+                          onPressed: () =>
+                              goTo(context, (_) => const InboxPage()),
```
[the remark this answers is not in the document]
Comment (zachatrocity) [reply lib/pages/home_tab.dart:257]: Perhaps, but we're not adding any new ways to go to the inbox yet, we should in the future. Will save that for a future PR

File: lib/pages/community/community.dart [hunk 4]
```diff
@@ -121,6 +121,7 @@ class CommunityPage extends HookWidget {
                 ),
               ],
               body: TabBarView(
+                physics: const NeverScrollableScrollPhysics(),
```
Comment (shocklateboy92) [inline lib/pages/community/community.dart:124]: Why is this included in the diff?
Isn't this change already in master?

File: lib/pages/home_tab.dart [hunk 5]
```diff
@@ -247,10 +247,21 @@ class HomeTab extends HookWidget {
...
                 titleTextStyle: theme.textTheme.titleLarge
                     ?.copyWith(fontSize: 20, fontWeight: FontWeight.w500),
-                leading: IconButton(
-                  icon: const Icon(Icons.notifications),
-                  onPressed: () => goTo(context, (_) => const InboxPage()),
-                ),
+                leading: accStore.notificationCount > 0
+                    ? Badge(
+                        offset: const Offset(-5, 5),
+                        label: Text(accStore.notificationCount.toString()),
```
Comment (shocklateboy92) [inline lib/pages/home_tab.dart:253]: I forgot to mention this in my original review:
Shouldn't we have a limit on this? 
I.e. render `9+` instead potentially letting this get to triple digits and looking really ugly?

File: lib/pages/inbox.dart [hunk 6]
```diff
@@ -61,32 +61,39 @@ class InboxPage extends HookWidget {
...
+                .expand(
+                  (instanceHost) => accStore
+                      .usernamesFor(instanceHost)
+                      .map((username) => '$username@$instanceHost'),
+                )
+                .toList(),
+            groupValue:
+                '${accStore.defaultUsername}@${accStore.defaultInstanceHost}',
+            onChanged: (value) {
+              final [user, instance] = value.split('@');
```
Comment (shocklateboy92) [inline lib/pages/inbox.dart:82]: Unrelated, but this particular snippet of code has appeared _a lot_ now. I think we should have an overload of `setDefaultAccount` that takes a full string and does the split internally.

File: lib/widgets/scoll_chainer.dart [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+import 'package:logging/logging.dart';
+
+final _logger = Logger('ScrollChainer');
+
+// Need to keep track of whether the user is scrolling or not, otherwise
+// the scroll animation will be overridden by the user's scroll.
+// Not using a stateful widget because it would cause an unnecessary rebuild
+// of the widget tree.
+// ignore: must_be_immutable
+class ScrollChainer extends StatelessWidget {
```
Comment (shocklateboy92) [inline lib/widgets/scoll_chainer.dart:11]: This is _definitely_ already in master.
Maybe rebase on latest master and see if everything still looks good before merging?

File: lib/widgets/user_profile.dart [hunk 8]
```diff
@@ -120,12 +107,30 @@ class UserProfile extends HookWidget {
                           ?.jwt
                           .raw,
                     ))
-                    .then((val) => val.comments),
+                    .then((val) => val.posts),
```
Comment (shocklateboy92) [inline lib/widgets/user_profile.dart:110]: Is this an intentional change? Or another bad merge?

**Review by shocklateboy92 — APPROVED** [review]

File: lib/widgets/user_profile.dart (hunk 8 above)
Comment (swmarks) [reply lib/widgets/user_profile.dart:110]: I think this might be what has broken the user profile view if you try to go between comments and posts.

Comment (swmarks) [reply lib/widgets/user_profile.dart:110]: If you try scrolling down the list of posts or comments, it does this weird snap back to the top thing.

File: lib/widgets/scoll_chainer.dart (hunk 7 above)
Comment (swmarks) [reply lib/widgets/scoll_chainer.dart:11]: Oh, I also just noticed a slight spelling error in the file name. Looks like scroll is missing an r

File: lib/widgets/user_profile.dart (hunk 8 above)
Comment (swmarks) [reply lib/widgets/user_profile.dart:110]: I believe the above described issue is actually caused by this: https://github.com/liftoff-app/liftoff/pull/287
